examples/handshake.rs

```diff
@@ -0,0 +1,102 @@
+use rand::Rng;
+use std::io::Write;
+use std::net::{IpAddr, Ipv4Addr, Shutdown, SocketAddr, TcpStream};
+use std::time::{SystemTime, UNIX_EPOCH};
+
+use bitcoin::consensus::encode;
+use bitcoin::network::address;
+use bitcoin::network::constants;
+use bitcoin::network::message;
+use bitcoin::network::message_network;
+use bitcoin::network::stream_reader::StreamReader;
+
+fn main() {
+    // This example establishes a connection to a Bitcoin node, sends the intial
+    // "version" message, waits for the reply, and finally closes the connection.
+
+    let address = SocketAddr::new(IpAddr::V4(Ipv4Addr::new(130, 149, 80, 221)), 8333);
+
+    let version_message = build_version_message(address.clone());
+
+    let first_message = message::RawNetworkMessage {
+        magic: constants::Network::Bitcoin.magic(),
+        payload: version_message,
+    };
+
+    if let Ok(mut stream) = TcpStream::connect(address) {
+        // Send the message
+        let _ = stream.write(encode::serialize(&first_message).as_slice());
```

Comment:
maybe use `write_all` instead? I don't know enough about TCP streams though

Reply:
I think it doesn't make a difference here, but since `write_all` would spare us the error handling, it is probably cleaner here. Thanks!